tshalvi/sonic-buildimage
# Independent Module - sonic-buildimage code to support port SI configuration per speed

#### Why I did it
In order to enable module configuration by SONiC, it's necessary to ignore the platform code that prevents writing to the EEPROM if the platform supports this functionality and the module needs to be configured by the NOS.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
I used the 'TRANSCEIVER_MODULES_MGMT' table in STATE_DB to determine whether the module needs to be configured by the NOS. Additionally, I checked the 'sai.profile' file to verify if the platform supports the required functionality.

#### How to verify it
You can verify this by ensuring that when SONiC attempts to configure an active CMIS module, you do not encounter a 'write limited bytes' exception.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
